src/Examine.AzureDirectory/AzureDirectory.cs

```diff
@@ -299,11 +375,9 @@ public override IndexInput OpenInput(string name)
             }
             else
             {
-                // Lucene expects this exception to be thrown.
-                // Except! not all of the time. When committing to the index the CompoundFileWriter will use this method:
-                // https://github.com/apache/lucenenet/blob/3.0.3/src/core/Index/CompoundFileWriter.cs#L241
-                // which cannot throw but there's not much to do about that and it should not occur.
-                throw new FileNotFoundException(name, err);
+                SetDirty();
+                return CacheDirectory.OpenInput(name);
+                //   throw new FileNotFoundException(name, err);
```

Comment:
Are you sure you want to remove this throw? What will happen if the local file doesn't open, will this exception just be thrown anyways?

Reply:
@Shazwazza that throw is happening when file is still in cache and isn't in blob, which suggest the sync process was started and you dont want serve 500 error as sync is happening in time of synchronization. And yes if local file doesnt exists it will throw same error.